### PR TITLE
update linear homogenization example

### PR DESCRIPTION
**Pull request by vlukes**

### CODE REVIEW

File: examples/homogenization/linear_homogenization_up.py [hunk 1]
```diff
@@ -151,32 +147,23 @@ def recovery_le( pb, corrs, macro ):
...
-    val = pis.states[ir, ic]['u'] + corrs_rs.states[ir, ic]['u']
-
-    variables[mode2var[mode]].set_data(val)
-
 coefs = {
     'elastic_u' : {
         'requires' : ['pis', 'corrs_rs'],
-        'expression' : expr_coefs['Q1'],
-        'set_variables' : set_elastic_u,
+        'expression' : 'dw_lin_elastic.i.Y( mat.D, Pi1u, Pi2u )',
```
Comment (rc) [inline examples/homogenization/linear_homogenization_up.py]: Remove extra spaces in the expression.

File: examples/homogenization/linear_homogenization_up.py [hunk 2]
```diff
@@ -151,32 +147,23 @@ def recovery_le( pb, corrs, macro ):
...
+        'set_variables' : [('Pi1u', ('pis', 'corrs_rs'), 'u'),
+                           ('Pi2u', ('pis', 'corrs_rs'), 'u')],
         'class' : cb.CoefSymSym,
     },
     'elastic_p' : {
         'requires' : ['corrs_rs'],
-        'expression' : expr_coefs['Q2'],
-        'set_variables' : [('Pi1p', 'corrs_rs', 'p'), ('Pi2p', 'corrs_rs', 'p')],
+        'expression' : 'dw_volume_dot.i.Y( mat.gamma, Pi1p, Pi2p )',
```
Comment (rc) [inline examples/homogenization/linear_homogenization_up.py]: Remove extra spaces in the expression.

File: examples/homogenization/linear_homogenization_up.py [hunk 3]
```diff
@@ -207,9 +194,11 @@ def set_elastic_u(variables, ir, ic, mode, pis, corrs_rs):
...
 #! Define linear and nonlinear solver.
 solvers = {
-    'ls' : ('ls.scipy_direct', {}),
+    'ls': ('ls.scipy_iterative', {
+        'method': 'cg',
+    }),
     'newton' : ('nls.newton', {
         'i_max' : 1,
-        'eps_a' : 1e-4,
+        'eps_a' : 1e2,
```
Comment (rc) [inline examples/homogenization/linear_homogenization_up.py]: Why is this necessary?

**Comment by rc** [comment]
Thanks! The changes look fine, only the comments above need to be addressed.

**Comment by vlukes** [comment]
The comments have been incorporated.

"'eps_a' : 1e2" has been changed to avoid the warning of a low precision solution. The correctors are of magnitude 1e9 so the tolerance 1e2 seems to be reasonable.

**Comment by rc** [comment]
@vlukes Or, you could set ``'lin_red' : 1e6,`` with the same effect. Anyway, your change is OK.

**Comment by rc** [comment]
@vlukes does the update require #352, or this error is unrelated?

```
./homogen.py examples/homogenization/linear_homogenization.py
...
  File "/home/user/projects/sfepy-git/sfepy/homogenization/homogen_app.py", line 19, in __call__
    problem.select_variables(self.variables)
AttributeError: 'Volume' object has no attribute 'variables'
```

**Comment by vlukes** [comment]
... yes, it requires #352

**Comment by rc** [comment]
OK, so I will merge this after #352 is finished and merged.
